feat(rxRadio): add rxRadio component

CITguy opened the pull request:
Per necessity of #884, this directive can stand alone for use elsewhere
in the framework.

CITguy commented:
*screenshot*
![rxradio-local-20150508-1](https://cloud.githubusercontent.com/assets/545605/7547280/cd6c9efa-f5af-11e4-823c-425c57049f97.png)

CITguy commented:
*two visible options, one hidden (all invalid)*
![screen shot 2015-05-08 at 6 25 06 pm](https://cloud.githubusercontent.com/assets/545605/7547284/dd9b508c-f5af-11e4-94af-ebf6989c5b16.png)

*first visible option selected, one still hidden (all valid)*
![screen shot 2015-05-08 at 6 25 13 pm](https://cloud.githubusercontent.com/assets/545605/7547285/dd9b5438-f5af-11e4-9060-abad4385c3b1.png)

*second visible option selected, triggers visibility of third option (all valid)*
![screen shot 2015-05-08 at 6 25 20 pm](https://cloud.githubusercontent.com/assets/545605/7547283/dd99f16a-f5af-11e4-8a9e-f42516fdc96a.png)

*previously hidden, third option selected (all valid)*
![screen shot 2015-05-08 at 6 25 28 pm](https://cloud.githubusercontent.com/assets/545605/7547282/dd990ba6-f5af-11e4-8003-2e17f5972242.png)

CITguy commented:
*target radio exists in DOM*
![screen shot 2015-05-08 at 6 33 53 pm](https://cloud.githubusercontent.com/assets/545605/7547338/be29eba4-f5b0-11e4-9d03-f1d419a1f14f.png)

*target radio removed from DOM*
![screen shot 2015-05-08 at 6 33 46 pm](https://cloud.githubusercontent.com/assets/545605/7547337/be27a04c-f5b0-11e4-984c-d87280cf46e1.png)

CITguy commented:
New screenshot based on related feedback in #936.

* automatically set `type="radio"` on input
* use `compile` rather than `link`
* use angular element as wrapper for DOM manipulation

Additionally, I reorganized documentation to make it easier to consume. All examples still work as shown in previous screenshots.

![rxradio-local-20150511-1](https://cloud.githubusercontent.com/assets/545605/7574713/0dc47940-f7f5-11e4-8dfe-dcadc58f9b44.png)

parlarjb commented:
Could you put back in .spec.js to ensure that type="checkbox" is being correctly added, as well as checking that your custom ng-disabled logic is being applied?

CITguy commented:
@parlarjb: I've added exercises and additional tests to cover the `ng-disabled` and `type="radio"` logic in the DDO.

parlarjb commented:
LGTM

megalord commented on src/rxRadio/docs/rxRadio.html:
```diff
@@ -0,0 +1,161 @@
+<!-- Sample HTML goes here as a live example of how the component can be used -->
+<div ng-controller="rxRadioCtrl">
+
+  <h2 class="title">Examples</h2>
```
I don't know if it's documented anywhere, but I think four spaces are usually used for html.

CITguy replied:
We currently have the following documented: 
* Scripts (JS): ["4 spaces for indentation"](https://github.com/rackerlabs/encore-ui/blob/master/guides/js-styleguide.md#basic-formatting)
* Styles (CSS/LESS): ["one level of indentation"](https://github.com/rackerlabs/encore-ui/blob/master/guides/css-styleguide.md#format)
* Markup (HTML): nothing documented

This is something that needs discussed amongst other developers and stakeholders. Please create an issue to document this concern.

megalord commented on src/rxRadio/rxRadio.less:
```diff
@@ -0,0 +1,101 @@
+    }
+  }//.fake-radio
+
+  &.rx-disabled + label {
```
So I know that issue #948 will probably address this, but there is an assumption here that the label is directly after the `<input>`.  I think it's worthwhile putting a line in the documentation about it, even if it changes as a result of the work for that issue.

CITguy replied:
Good point. I'll put some documentation around that.

CITguy replied:
After I think about it, I feel like we should hold off until a decision has been made in #948. For the initial release of this directive, we'll support `input + label`.

megalord commented on src/rxRadio/rxRadio.js:
```diff
@@ -0,0 +1,39 @@
+
+                // apply/remove disabled attribute so we can
+                // apply a CSS selector to style sibling elements
+                scope.$watch('ngDisabled', function (newVal) {
```
Though I'm not sure if there's a use case out there, this won't support the native `disabled` attribute.  Maybe it doesn't really matter - just thought I'd point it out.

CITguy replied:
This is a valid point. I overlooked the native `disabled` attribute in the logic. I'll put something in to correct this.

megalord commented on src/rxRadio/rxRadio.spec.js:
```diff
@@ -0,0 +1,21 @@
+        el = helpers.createDirective(validTemplate, compile, scope);
+    });
+
+    /* reserve for potential future specs */
```
Is this going to get the same treatment as `rxCheckbox`?

CITguy replied:
Huh, I thought I removed this file. Yes, it will only have midways, like `rxCheckbox`.

CITguy commented:
* removed unused `spec.js` file
* added support for native `disabled` attribute

*Radio States - updated to include native `disabled` examples*
![screen shot 2015-05-13 at 10 55 38 am](https://cloud.githubusercontent.com/assets/545605/7614918/a7f91e10-f95f-11e4-88ba-4410dafd5661.png)

parlarjb commented on src/rxRadio/rxRadio.js:
```diff
@@ -0,0 +1,45 @@
+                if (element.attr('disabled')) {
+                    wrapper.addClass(disabledClass);
+                }
+                if (scope.ngDisabled) {
```
Same issue as the other PR

CITguy commented:
Updated to use `if (_.has(attrs, 'ngDisabled'))`

megalord commented:
LGTM
